### PR TITLE
Add pattern_capture token filter docs #8240

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add pattern_capture token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8240

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
> @udabhas Could you review this PR for technical accuracy?

@udabhas @varun-lodaya This is almost a month old. We need tech review approval to move it forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/token-filters/pattern-capture.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+The `pattern_capture` token filter is a powerful filter that uses regular expressions to capture and extract parts of text according to specific patterns. This filter can be useful when you want to extract particular parts of tokens, such as email domains, hashtags, or numbers, and reuse them for further analysis or indexing.
+
+## Parameters
+
+The `pattern_capture` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`patterns` | Required | Array of strings | An array of regular expressions used to capture parts of the text.
```
Comment (natebower) [inline _analyzers/token-filters/pattern-capture.md:18]:
```suggestion
`patterns` | Required | Array of strings | An array of regular expressions used to capture parts of text.
```

File: _analyzers/token-filters/pattern-capture.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+`preserve_original` | Required | Boolean| Whether to keep the original token in the output. Default is `true`.
+
+
+## Example
+
+The following example request creates a new index named `email_index` and configures an analyzer with a `pattern_capture` filter to extract the local part and domain name from an email address:
```
Comment (natebower) [inline _analyzers/token-filters/pattern-capture.md:24]: Is "part" the right word here?

Comment (kolchfa-aws) [reply _analyzers/token-filters/pattern-capture.md:24]: Yup, it's the proper term https://en.wikipedia.org/wiki/Email_address

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**